packages/bvaughn-architecture-demo/src/suspense/PauseCache.ts

```diff
@@ -54,12 +56,18 @@ export const {
   (pauseId, frameId, expression) => `${pauseId}:${frameId}:${expression}`
 );
 
+export function getCachedCallStackForPauseId(pauseId: PauseId): CallStack | null {
+  return callStacksByPauseId.get(pauseId) || null;
+}
+
 export function cachePauseData(
   client: ReplayClientInterface,
   pauseId: PauseId,
   pauseData: PauseData,
   stack?: CallStack
 ) {
+  callStacksByPauseId.set(pauseId, stack || null);
```

Comment:
The `stack` should only be cached if it comes from `Session.createPause` or `Pause.getAllFrames`, there are other messages that can contain `PauseData` and don't provide a `stack` (even if it isn't empty for that pauseId).

Reply:
Can we just not pass the `stack` param in those cases then? (It's optional.)